### ISSUE TITLE
[java] JUnitTestsShouldIncludeAssert: Support AssertJ soft assertions

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#1435]
**Rule:** [JUnitTestsShouldIncludeAssert](https://pmd.github.io/pmd-6.9.0/pmd_rules_java_bestpractices.html#junittestsshouldincludeassert)

**Description:**

Using [Soft Assertions](https://github.com/joel-costigliola/assertj-core) shouldn't trigger the rule, since these are assertions, too.

**Code Sample demonstrating the issue:**

```java
import org.assertj.core.api.SoftAssertions;
import org.junit.jupiter.api.Test;
class FooTest {
    @Test
    void testFoo() {
        var softly = new SoftAssertions();
        softly.assertThat("doesn't matter").isEqualTo("doesn't matter");
        softly.assertAll();
    }
```
